heudiconv/tests/test_bids.py

```diff
@@ -46,6 +50,13 @@
 
 import pytest
 
+def gen_rand_label(label_size, label_seed):
+    np.random.seed(label_seed)
+    rand_char = ''.join(choice(string.ascii_letters) for _ in range(label_size-1))
+    np.random.seed(label_seed)
+    rand_num = choice(string.digits)
```

Comment:
Since `choice` is part of a different module (`random`), does `np.random.seed` have an effect here?
Also, do we need to initialize the seed? I think just generating random labels is enough, without the need to be able to reproduce them, isn't it?
If you are OK with it, I would just drop the two `np.random.seed` calls (and the `import numpy`)

Reply:
@pvelasco you are completely right, it should be the seed from the `random` library !

The seed is there to avoid a "heisenbug", i.e. a test crash which could disappear once we try looking into it (because then the random label would be different). It also allows to rerun tests several times with different seeds (while still be able to investigate a crash associated to a specific random instance).

So if you prefer to remove randomness, a fixed string may be better than removing the seed. I don't mind reverting to a fixed string if you think that's better. Otherwise i'll replace the call to `np.random.seed` with `random.seed` (and remove the `import numpy`).

Reply:
Meanwhile i corrected the seeding with 457bf406a4f2ce4cad39963ff697ba7d201c1111

Reply:
> The seed is there to avoid a "heisenbug", i.e. a test crash which could disappear once we try looking into it (because then the random label would be different). It also allows to rerun tests several times with different seeds (while still be able to investigate a crash associated to a specific random instance).

OK. That makes sense. However, if I read the code correctly, it will always use the same seed (`42`). Doesn't that mean that the labels generated (`A_LABEL`, `DWI_LABEL`, etc.) will always be the same? If so, the tests will still always check the same values.

Would it be possible just to pick the default value for `label_seed` (in `create_dummy_no_shim_settings_custom_label_bids_session` and `test_get_key_info_for_fmap_assignment`) at random, and then print it out to the std out, and then pass it to `gen_rand_label`? That way every time we run the tests, it will use different random labels, and because the seed is printed in the output, we can replicate them if they fail.

Something like:
```
def create_dummy_no_shim_settings_custom_label_bids_session(session_path, label_size=4, label_seed=randrange(100)):
print(f'label_seed: {label_seed}')
```

Reply:
The seed is by default the same indeed. The idea is to provide an array of seeds as parametrization to `pytest` when wanting to test different random labels.

Meanwhile i implemented in 547dafd30d83a0f30ebb8be2697082e4f2811f6c something along the lines of what you suggested, with a single test suite seed which will change at every run of pytest and which is printed to stdout.

Reply:
Great!
Thank you, @neurorepro